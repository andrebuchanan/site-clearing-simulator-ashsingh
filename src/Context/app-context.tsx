import React, { ReactNode, useMemo, useState } from 'react'

type Session = {
  init: boolean
  quit: boolean
}

interface AppContextInterface {
  fuelConsumed: number[]
  itemizedCosts: number[]
  loggedCommands: string[]
  sessionState: Session
  siteData: string[][]
  logCommand: (command: string) => void
  saveSiteData: (data: string[][]) => void
  updateCost: (price: number) => void
  updateFuel: (quantity: number) => void
  updateSession: (session: Session) => void
}

export const AppContext = React.createContext<AppContextInterface>({
  fuelConsumed: [],
  itemizedCosts: [],
  loggedCommands: [],
  sessionState: { init: true, quit: false },
  siteData: [],
  logCommand: () => undefined,
  saveSiteData: () => undefined,
  updateCost: () => undefined,
  updateFuel: () => undefined,
  updateSession: () => undefined
})

export const AppContextProvider = ({ children }: { children: ReactNode }) => {
  const [commands, setCommands] = useState<string[]>([])
  const [cost, setCost] = useState<number[]>([])
  const [fuel, setFuel] = useState<number[]>([])
  const [session, setSession] = useState<Session>({ init: true, quit: false })
  const [siteInput, setSiteInput] = useState<string[][]>([])

  const logCommandHandler = (command: string) => {
    setCommands((prevState) => [...prevState, command])
  }

  const saveSiteDataHandler = (data: string[][]) => {
    setSiteInput(data)
  }

  const updateCostHandler = (price: number) => {
    setCost((prevState) => [...prevState, price])
  }

  const updateFuelHandler = (quantity: number) => {
    setFuel((prevState) => [...prevState, quantity])
  }

  const updateSessionHandler = (sessionProp: Session) => {
    console.log('session', sessionProp)
    if (sessionProp.quit) console.log('quit')
    setSession(sessionProp)
  }

  const contextValue = useMemo(
    () => ({
      fuelConsumed: fuel,
      itemizedCosts: cost,
      loggedCommands: commands,
      sessionState: session,
      siteData: siteInput,
      logCommand: logCommandHandler,
      saveSiteData: saveSiteDataHandler,
      updateCost: updateCostHandler,
      updateFuel: updateFuelHandler,
      updateSession: updateSessionHandler
    }),
    [commands, cost, fuel, session, siteInput]
  )

  return (
    <AppContext.Provider value={contextValue}>{children}</AppContext.Provider>
  )
}

export default AppContextProvider
