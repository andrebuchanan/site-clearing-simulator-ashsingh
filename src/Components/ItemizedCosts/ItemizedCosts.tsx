import React, { useContext } from 'react'
import uniqid from 'uniqid'

import List from '@mui/material/List'
import ListItem from '@mui/material/ListItem'
import ListItemText from '@mui/material/ListItemText'
import Typography from '@mui/material/Typography'

import { AppContext } from '../../Context'

export const ItemizedCosts = () => {
  const { itemizedCosts } = useContext(AppContext)
  return (
    <>
      <Typography variant="h6" component="div" gutterBottom>
        Itemized Costs
      </Typography>
      <List>
        {itemizedCosts.map((cost) => (
          <ListItem key={uniqid()} disablePadding>
            <ListItemText primary={`${cost} credits`} />
          </ListItem>
        ))}
      </List>
    </>
  )
}

export default ItemizedCosts
