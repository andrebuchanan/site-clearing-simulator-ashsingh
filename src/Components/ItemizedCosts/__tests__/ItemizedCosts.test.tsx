import React from 'react'
import { render, screen } from '@testing-library/react'

import { AppContext } from '../../../Context'
import ItemizedCosts from '../ItemizedCosts'

const contextValue = {
  itemizedCosts: [0, 0, 7]
}

test('renders App', () => {
  const { asFragment } = render(
    <AppContext.Provider value={contextValue}>
      <ItemizedCosts />
    </AppContext.Provider>
  )

  expect(asFragment()).toMatchSnapshot()
  expect(screen.getByText('7 credits')).toBeDefined()
})
