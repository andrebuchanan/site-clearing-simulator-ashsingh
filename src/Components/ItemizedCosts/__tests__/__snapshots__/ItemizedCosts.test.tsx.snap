// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<DocumentFragment>
  <div
    class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
  >
    Itemized Costs
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-11jh7f1-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <div
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-divider MuiListItem-button css-1ghby9w-MuiButtonBase-root-MuiListItem-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            0 credits
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-divider MuiListItem-button css-1ghby9w-MuiButtonBase-root-MuiListItem-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            0 credits
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-divider MuiListItem-button css-1ghby9w-MuiButtonBase-root-MuiListItem-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            7 credits
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </ul>
  </div>
</DocumentFragment>
`;
